lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -1073,6 +1066,23 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
       }
     }
 
+    SmallVector<Value> inputSize = getTensorSizes(rewriter, loc, input);
+
+    SmallVector<Value> outputSizeInt = getTypeConvertedValues(
+        rewriter, loc, typeConverter, outputSizeTorchInt);
+    if (resultRank != (int64_t)outputSizeInt.size()) {
+      return rewriter.notifyMatchFailure(
+          op, "desired size list length mismatches with the result type rank");
+    }
+
+    for (auto pair : checkDimPairs) {
```

Comment:
https://en.cppreference.com/w/cpp/language/structured_binding

Can we use

auto [inputDim, outputDim] : checkDimPairs

to make this more readable? Or does it only work for `=` and not `:`

Reply:
I somehow knew someone would call me out on this. Darn C++ 21. Fixed.